CON-4414: Audit API documentation has been updated to include info on events/ path query strings

izgeri opened the pull request:
closes CON-4414

updating audit documentation to:
- include info on limit/since/till
- add documentation for events/ path with query string
- update 404 error to be `not found / permission denied` since we sometimes return this error for a permissions issue

Commit:
updating audit documentation
- include info on limit/since/till
- add documentation for events/ path with query string
- update 404 error to be `not found / request refused`

Commit:
change language from request refused to permission denied

Commit:
update array query param syntax to match what's used in host factory

kgilpin reviewed (commented):
The changes look good; something is wrong in Jenkins though - https://jenkins.conjur.net/job/conjurinc--apidocs/job/CON-4414-audit-api-documentation-updated-event-path/

kgilpin commented on src/audit.single.md:
```diff
@@ -17,16 +17,20 @@ Fetch audit events for a role/resource the calling identity has `read` privilege
-    + kind: roles (string) - Type of object, 'roles' or 'resources'
-    + id: cucumber:host:redis001 (string) - Fully qualified ID of a Conjur role/resource, query-escaped
+    + kind: `roles` (string) - Type of object, 'roles' or 'resources'
+    + id: `cucumber:host:redis001` (string) - Fully qualified ID of a Conjur role/resource, query-escaped
```
We say it must be query escaped, but in the example it isn't. @dustinmm80 and I have looked at this problem and unfortunately we can't find a way to fix it.

Commit:
update transactions file for tests to include new labels / new doc file

kgilpin commented on src/audit.multiple.md:
```diff
@@ -0,0 +1,88 @@
+
+### Fetch audit events matching any subset of roles or resources [GET]
+
+Fetch audit events matching any of a set of roles and/or resources the calling identity has `read` privilege on.
```
Does that mean that all the roles and resources are OR-ed together? Can you describe the use case of why this is useful? While I am reading the doc, it would be helpful to describe the scenario in which I would use this, rather than the routes that filter on a single role or resource.

kgilpin commented on src/audit.multiple.md:
```diff
@@ -0,0 +1,88 @@
+## Multiple Roles or Resources [/api/audit/events{?role%5B%5D,resource%5B%5D,event_action,kind,limit,till,since}]
```
Since this is an existing route, shouldn't this doc enhance the description of that route?

Commit:
Updating README, minor mod to audit docs
- give details on summon install for running tests
- update audit / multiple docs to use valid resource so tests will pass

izgeri commented:
@kgilpin it took me a minute to get my dev environment configured correctly to determine the root cause of the Jenkins test failure, but the status is all green now.

Heading over to the gdoc on this topic to contribute more info now ...

izgeri commented:
The audit API updates are not being merged at this time; in light of this, I'm closing this PR until such time as these updates become part of the system (if ever).
